src/workerd/io/compatibility-date.capnp

```diff
@@ -297,4 +297,11 @@ struct CompatibilityFlags @0x8f8c1b68151b6cef {
   # This one operates a bit backwards. With the flag *enabled* no default cfBotManagement
   # data will be included. The the flag *disable*, default cfBotManagement data will be
   # included in the request.cf if the field is not present.
+
+  urlSearchParamsDeleteHasValueArg @30 :Bool
+      $compatEnableFlag("urlsearchparams_delete_has_value_arg")
+      $compatDisableFlag("no_urlsearchparams_delete_has_value_arg")
+      $compatEnableDate("2023-07-01");
```

Comment:
General question for all reviewers, how do we determine when we want the date to be? Also is there any reason we don't want to set this compat date sooner?

Reply:
It's fairly arbitrary and case-by-case really. I'll usually try to go at least a month out.